examples/03_connectivity/plot_confounds_removal.py

```diff
@@ -0,0 +1,170 @@
+"""
+Carefully defining the confounds
+================================
+
+This example shows the importance of confounds removal for functional
+connectivity.
+
+"""
+########################################
+# We use one ADHD subject.
+from nilearn import datasets
+adhd = datasets.fetch_adhd(n_subjects=1)
+func_filename = adhd.func[0]
+
+######################################
+# Friston 24 motion parameters
+# -------------------------------------
+# They include the 6 motion parameters of the current volume
+# and the preceding volume, plus each of these values squared.
+
+# Read the 6 motion parameters from the confounds file
+import numpy as np
+csv_confounds = np.recfromcsv(adhd.confounds[0], delimiter='\t',
+                              usecols=('motionx', 'motiony', 'motionz',
+                                       'motionpitch', 'motionroll',
+                                       'motionyaw'))
+current_volume_motion = csv_confounds.view(dtype=float).reshape(-1, 6)
+previous_volume_motion = np.vstack((current_volume_motion[0],
+                                    current_volume_motion[:-1]))
+confounds = np.hstack((current_volume_motion, previous_volume_motion,
+                       current_volume_motion ** 2,
+                       previous_volume_motion ** 2))
+
+##########################################################################
+# Anatomical CompCor
+# ----------------------------------------------
+# Signals within noisy ROIs, such as WM and CSF, are assumed to reflect noise.
+# The anatomical CompCor approach removes the first principal components
+# of noisy ROIs.
+#
+# First, we define the tissue masks.
+# As we do not have an anatomical image, we are relying on group-level masks.
+# Note that subject-specific masks would give more accurate results.
+
+##########################################################################
+# We give the locations of probabilistic group masks
+icbm152 = datasets.fetch_icbm152_2009()
+mask_images = {'gm': icbm152.gm, 'wm': icbm152.wm, 'csf': icbm152.csf}
+
+##########################################################################
+# and use them to compute binary masks.
+from nilearn import image
+for tissue_name, threshold in zip(['wm', 'gm', 'csf'], ['.9', '.5', '.3']):
```

Comment:
Will it be possible to add more narration here ?

We use ```math_img``` from nilearn to convert probabilistic masks to binary masks with threshold being adjusted manually or something similar.